Blazor: add support for ontoggle event

SamProf opened the pull request:
Summary of the changes
- Implemented `@ontoggle` event
- Added test component to BasicTestApp

Fix Issue: #20859

Question: Should I commit also this files? I am not sure because they can be regenerated after build.
- `\src\Components\Web.JS\dist\Release\blazor.server.js`
- `\src\Components\Web.JS\dist\Release\blazor.webassembly.js`

SamProf committed:
Blazor: add support for toggle event #20859
- added support of ontoggle event
- added test in BasicTestApp

SamProf committed:
Merge branch 'master' of https://github.com/SamProf/aspnetcore

SamProf committed:
Blazor: add comment and changed BasicTestApp example for toggle event #20859

dnfadmin commented:
[![CLA assistant check](https://cla.dotnetfoundation.org/pull/badge/not_signed)](https://cla.dotnetfoundation.org/dotnet/aspnetcore?pullRequest=24036) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/aspnetcore?pullRequest=24036) before we can accept your contribution.<br/><br/>:x: SamProf [sign now](https://cla.dotnetfoundation.org/dotnet/aspnetcore?pullRequest=24036)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/aspnetcore?pullRequest=24036) it.</sub>

pranavkm commented on src/Components/Web/src/Web/EventHandlers.cs:
```diff
@@ -122,6 +122,9 @@ namespace Microsoft.AspNetCore.Components.Web
     [EventHandler("onscroll", typeof(EventArgs), true, true)]
+
+    // Specific events
+    [EventHandler("ontoggle", typeof(EventArgs), true, true)]
```
You'll likely have to regenerate ref assemblies: https://github.com/dotnet/aspnetcore/blob/master/docs/ReferenceAssemblies.md#when-changing-public-api

pranavkm commented on src/Components/Web.JS/src/Rendering/EventForDotNet.ts:
```diff
@@ -89,6 +89,9 @@ export class EventForDotNet<TData extends UIEventArgs> {
         return new EventForDotNet<UIWheelEventArgs>('wheel', parseWheelEvent(event as WheelEvent));
 
+      case 'toggle':
+        return new EventForDotNet<UIEventArgs>('toggle', {type: event.type});
```
```suggestion
        return new EventForDotNet<UIEventArgs>('toggle', { type: event.type });
```

pranavkm commented on src/Components/Web/src/Web/EventHandlers.cs:
```diff
@@ -122,6 +122,9 @@ namespace Microsoft.AspNetCore.Components.Web
     [EventHandler("onreadystatechange", typeof(EventArgs), true, true)]
     [EventHandler("onscroll", typeof(EventArgs), true, true)]
+
+    // Specific events
```
This comment doesn't seem very descriptive. You could consider removing it

pranavkm commented on src/Components/test/testassets/BasicTestApp/EventCallbackTest/OnToggleEventTestComponent.razor:
```diff
@@ -0,0 +1,20 @@
+@code {
+    int ToggleCount { get; set; }
+
+    void OnToggle(EventArgs e)
```
Could you add a test that goes with it? It would go under the E2E directory.

pranavkm commented:
> Question: Should I commit also this files? I am not sure because they can be regenerated after build.

Yes please. We commit these to source for some hard-to-explain reasons.

SamProf committed:
Blazor: ontoggle event review changes #20859:
- Regenerated ref assemblies
- Formatting
- Removing not useful comment
- Commited blazor.server.js and blazor.webassembly.js

SamProf committed:
Blazor: add support for toggle event #20859 - E2E Tests

SamProf committed:
Merge branch 'master' of https://github.com/dotnet/aspnetcore

# Conflicts:
#	src/Components/Web.JS/dist/Release/blazor.server.js
#	src/Components/Web.JS/dist/Release/blazor.webassembly.js

SamProf committed:
Regenerate js files

SamProf commented:
@pranavkm Looks like I fix all your review propositions.

pranavkm commented:
Thanks for the PR!
